site/app/libraries/database/DatabaseQueries.php

```diff
@@ -4417,14 +4424,55 @@ public function mergeThread($parent_thread_id, $child_thread_id, &$message, &$ch
         return false;
     }
 
-    public function getAnonId($user_id) {
-        $params = (is_array($user_id)) ? $user_id : [$user_id];
+    /**
+     * Set gradeable-specific user anon_id
+     *
+     * @param array|string $user_ids
+     * @param string $g_id
+     * @param string $anon_id
+     */
+    public function insertGradeableAnonId($user_ids, $g_id, $anon_id = null) {
+        $user_ids = is_array($user_ids) ? $user_ids : [$user_ids];
+        foreach ($user_ids as $user_id) {
+            if ($anon_id == null) {
+                $this->core->getQueries()->getUserById($user_id)->getAnonId($g_id);
+                continue;
+            }
+            try {
+                $params = [$user_id, $g_id, $anon_id];
+                $this->course_db->query("INSERT INTO gradeable_anon(user_id, g_id, anon_id) VALUES (?, ?, ?)", $params);
+                $this->course_db->commit();
+            }
+            catch (DatabaseException $dbException) {
+                $this->course_db->rollback();
```

Comment:
Why commit and possibly do a rollback if you never start a transaction?

Reply:
Oh, it was left there when I copied the template from one of the functions above. This has been removed.